cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/OperationsDashboardHttpHandler.java

```diff
@@ -67,36 +70,30 @@ public OperationsDashboardHttpHandler(Store store) {
   public void readDashboardDetail(FullHttpRequest request, HttpResponder responder,
                                   @QueryParam("start") long startTimeSecs,
                                   @QueryParam("duration") int durationTimeSecs,
-                                  @QueryParam("namespace") Set<String> namespaces)
-    throws IOException, BadRequestException {
+                                  @QueryParam("namespace") Set<String> namespaces) throws BadRequestException {
     if (startTimeSecs < 0) {
       throw new BadRequestException("'start' time cannot be smaller than 0.");
     }
     if (durationTimeSecs < 0) {
       throw new BadRequestException("'duration' cannot be smaller than 0.");
     }
+    if (namespaces.isEmpty()) {
+      throw new BadRequestException("'namespace' cannot be empty, please provide atleast one namespace.");
+    }
 
-    Set<NamespaceId> namespaceIds = namespaces.stream().map(NamespaceId::new).collect(Collectors.toSet());
-    Map<ProgramRunId, RunRecordMeta> historicalRuns =
-      // TODO: [CDAP-13352] Currently, to get active program runs within a time range,
-      // a full table scan is required in AppMetaStore. Performance improvement will be done.
-      store.getHistoricalRuns(namespaceIds, startTimeSecs, startTimeSecs + durationTimeSecs, Integer.MAX_VALUE);
-    // get historical runs within the query time range
-    Stream<DashboardProgramRunRecord> historicalRecords =
-      historicalRuns.values().stream().map(OperationsDashboardHttpHandler::runRecordToDashboardRecord);
-    // get active runs with start time earlier than the end of query time range
-    Stream<DashboardProgramRunRecord> activeRecords =
-      store.getActiveRuns(namespaceIds, run -> run.getStartTs() < startTimeSecs + durationTimeSecs).values().stream()
-      .map(OperationsDashboardHttpHandler::runRecordToDashboardRecord);
-    responder.sendJson(HttpResponseStatus.OK,
-                       // combine historical runs and active runs
-                       GSON.toJson(Stream.concat(historicalRecords, activeRecords).collect(Collectors.toSet())));
+    Collection<RunRecordMeta> runRecordMetas =
+      programHeartbeatService.scan(startTimeSecs, startTimeSecs + durationTimeSecs + 1, namespaces);
+    List<DashboardProgramRunRecord> result =
+      runRecordMetas.stream()
+        .map(runRecordMeta -> runRecordToDashboardRecord(runRecordMeta)).collect(Collectors.toList());
```

Comment:
i think the closure in the map() can be replaced with a method reference (intellij marks this as a warning for me when a method reference can be used)